gobblin-runtime/src/main/java/gobblin/runtime/MultiConverter.java

```diff
@@ -80,82 +78,111 @@ public Object convertSchema(Object inputSchema, WorkUnitState workUnit)
   }
 
   /**
-   * A type of {@link java.util.Iterator} to be used with {@link MultiConverter}.
+   * A type of {@link java.util.Iterator} to be used with {@link MultiConverter}. This method works by maintaining a
```

Comment:
I think a lot simpler solution is to have a concept of the ChainedConverterIterator(Iterator prevIterator, Converter converter). Initialize it with the first value from prevIterator and convert it to a local iterator. Use this to return the initial values. Once the local iterator runs out of values, get the next value from prevIterator, convert it to a new local iterator and so on.